fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBDirectoryWrapper.java

```diff
@@ -111,7 +111,7 @@ public DirectoryReader getWriterReader(boolean flush) throws IOException {
         return writerReader;
     }
 
-    private static class FDBDirectoryMergeScheduler extends ConcurrentMergeScheduler {
+    private static class FDBDirectoryMergeScheduler extends SerialMergeScheduler {
```

Comment:
We should add a test that confirms the goals of switching to `SerialMergeScheduler`, and probably document why we want to switch.
Looking at logs/comments I could not figure out why we use the `ConcurrentMergeScheduler` to begin with, it appears to have part of the initial implementation commit ac71c207
Primarily, my understanding is that if you use `ConcurrentMergeScheduler` it will continue to do merges in the background, even after we have moved on. This may only happen in error scenarios. I came across it in https://github.com/FoundationDB/fdb-record-layer/pull/2569 but there is probably a more targeted test that we could write (that test is actually trying to catch a totall different bug).

Reply:
I can think of two potential ways to test this in a more targeted way:
1. Utilize the `FDBStoreTimer` and verify that after `explicitMerge` completes the timer doesn't change for some time
2. Fetch `Thread.getCurrenthread().getThreadGroup()` and validate that it is empty after `explicitMerge` completes. You may have to filter out threads like in `FDBThreadFilter`, but I suspect you could get a test that fails with the `ConcurrentMergeScheduler`. 

I think in order to get a useful test, you would need to have some more interesting merges; I think having a loop that saves 2 records/transaction with deferred merge, and then doing a merge with `LuceneRecordContextProperties.LUCENE_MERGE_SEGMENTS_PER_TIER` set to something low (like 3.0) should get enough merges that the `ConcurrentMergeScheduler` ends up having a lot of work.